packages/block-editor/src/components/block-variation-transforms/index.js

```diff
@@ -84,4 +94,82 @@ function __experimentalBlockVariationTransforms( { blockClientId } ) {
 	);
 }
 
+function __experimentalBlockVariationTransforms( { blockClientId } ) {
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+	const { variations, blockAttributes, blockName } = useSelect(
+		( select ) => {
+			const { getBlockVariations } = select( blocksStore );
+			const { getBlockName, getBlockAttributes } = select(
+				blockEditorStore
+			);
+			const name = blockClientId && getBlockName( blockClientId );
+			return {
+				blockAttributes: getBlockAttributes( blockClientId ),
+				blockName: name,
+				variations: name && getBlockVariations( name, 'transform' ),
+			};
+		},
+		[ blockClientId ]
+	);
+	const { activeBlockVariation } = useSelect(
+		( select ) => {
+			const { getActiveBlockVariation } = select( blocksStore );
+			return {
+				activeBlockVariation: getActiveBlockVariation(
+					blockName,
+					blockAttributes
+				),
+			};
+		},
+		[ blockAttributes, blockName ]
+	);
+
+	const selectedValue = activeBlockVariation?.name;
+
+	// Check if each variation has a unique icon.
+	const hasUniqueIcons = useMemo( () => {
+		const variationIcons = new Set();
+		variations.forEach( ( variation ) => {
+			if ( variation.icon ) {
+				variationIcons.add( variation.icon );
+			}
+		} );
+		return variationIcons.size === variations.length;
+	}, [ variations ] );
+
+	const onSelectVariation = ( variationName ) => {
+		updateBlockAttributes( blockClientId, {
+			...variations.find( ( { name } ) => name === variationName )
+				.attributes,
+		} );
+	};
+
+	const baseClass = 'block-editor-block-variation-transforms';
+
+	// Skip rendering if there are no variations
+	if ( ! variations?.length ) return null;
+
+	// If each variation has a unique icon, then render the variations as a set of buttons.
+	if ( hasUniqueIcons ) {
```

Comment:
Nit: You could write something like this: `const Component = hasUniqueIcons ? VariationsButtons : VariationsDropdown;` and then always return `<Component.......`

Reply:
Good catch! Thanks for the tip, I've consolidated into `Component` in https://github.com/WordPress/gutenberg/pull/40036/commits/eebbba78b7d94c1ae2b38f12d75bfd8500d867d8